### PR TITLE
Add react app

### PR DESCRIPTION
**Pull request by alicemirigo92**
- [ ] Create a new Rails API app called 'hello-rails-back-end'.
- [ ] Initialize your project with Git.
- [ ] Make sure that your project has a Postgres database set up.
- [ ] Create a table for storing your messages and create 5 different greetings.
- [ ] Create an API endpoint that selects a random greeting from your table (you will need a controller with an action and rails route).
- [ ] Create a new React app called 'hello-react-front-end'
- [ ] Initialize your project with Git.
- [ ] Create the App component with react-router.
- [ ] Create the Greeting component that will display a greeting. Set it up as a route in your App component.
- [ ] Create a store, an action and a reducer that will connect to you API endpont to get the random greeting.
- [ ] Display the random greeting in your Greeting component.

### CODE REVIEW

**Review by zainsadaqat — COMMENTED** [review]
# ❌ ❌ ❌Invalid Code Review Request ❌ ❌ ❌
# Hello @alicemirigo92  👋  

## Something went wrong 

### While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

Your Code Review Request is marked as invalid for this reason:

- ### You have submitted a React Project for the review However you were supposed to submit the link to the Rails API project as you can see in the screenshot of Project Requirements. Don't worry it won't take you much time. 👍 
Screenshot 2023-06-18 at 12 01 04 AM


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers, and Happy coding!👏👏👏



Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by franclobo — COMMENTED** [review]
Hi @alicemirigo92, 

# Invalid Code Review Request

descarga

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted the wrong link for the PR, to avoid misunderstanding please 🙏🏽 read the instructions 👇🏽 

## When submitting for a review - use a link to the pull request in your Rails API.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/App.js [hunk 1]
```diff
@@ -1,24 +1,30 @@
...
+      <header>
+        <nav>
+          <NavLink to="/Greeting" onClick={greet}>Greetings</NavLink>
+        </nav>
       </header>
-    </div>
+      <Routes>
+        <Route path="/Greeting" element={<Greeting />} />
+      </Routes>
+    </>
```
Comment (Owusu-Desmond) [inline src/App.js:27]: - [ ] Please from the project requirement, Your app should render random greeting on the page but currently no greetings is rendering on the page as you can see below :point_down: 

Screenshot from 2023-06-24 16-19-18

File: src/components/Greeting.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+
+function Greeting() {
+  const greeting = useSelector((state) => state);
+  return (
+    <h2>{greeting.message}</h2>
```
Comment (Owusu-Desmond) [inline src/components/Greeting.js]: - [ ] I have found out that you are calling the wrong property on the greeting api and I think changing `greeting.message` to `greeting.text` will fix the issue mentioned above :+1: 

image

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @alicemirigo92 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
📌 Descriptive pull request✅
📌 Great working on the set up of the project ✅
📌 Good commit messages ✅

## Required Changes ♻️

- [ ] Please make sure that you remove this broken link from your readme :point_down: 
  image

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Greeting.js (hunk 2 above)
Comment (alicemirigo92) [reply src/components/Greeting.js]: @Owusu-Desmond Thanks for review. I implemented the required changes

**Review by franclobo — APPROVED** [review]
Hi @alicemirigo92, 

# STATUS: APPROVED 😃 

git-merge

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct Git flow ✔️ 
- API is working well ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**